go/obsclient/authclient.go

```diff
@@ -62,9 +64,11 @@ func (ac *AuthObsClient) TransactionReceipt(ctx context.Context, txHash common.H
 	return &receipt, err
 }
 
-func (ac *AuthObsClient) NonceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (uint64, error) {
+// NonceAt retrieves the nonce for the account registered on this client (due to obscuro privacy restrictions,
+// nonce cannot be requested for other accounts)
+func (ac *AuthObsClient) NonceAt(ctx context.Context, blockNumber *big.Int) (uint64, error) {
 	var result hexutil.Uint64
-	err := ac.RPCClient.CallContext(ctx, &result, rpc.RPCNonce, account, toBlockNumArg(blockNumber))
+	err := ac.RPCClient.CallContext(ctx, &result, rpc.RPCNonce, ac.Account.Hex(), toBlockNumArg(blockNumber))
```

Comment:
@BedrockSquirrel Why have we switched from passing a `common.Address` to a `string` here?